cMake/SetupPETSC.cmake

```diff
@@ -0,0 +1,42 @@
+# Locate PETSc include paths and libraries
+# PETSc can be found at https://gitlab.com/petsc/petsc
+
+# This module defines
+# PETSC_INCLUDE_DIRS, where to find petsc.h, etc.
+# PETSC_LIBRARIES, the libraries to link against to use petsc.
+# PETSC_FOUND, If false, don't try to use petsc.
+
+function(ConfigurePETSC TARGET)
+  message(STATUS "Looking for PETSc")
+
+  # FIXME: windows paths were not tested
+  FIND_PATH(PETSC_INCLUDE_PATH
+    NAMES petsc.h
+    PATHS
+    ${CMAKE_INCLUDE_PATH}
+    "[HKEY_CURRENT_USER\\software\\petsc\\include]"
+    "[HKEY_CURRENT_USER\\petsc\\include]"
+    $ENV{PETSCROOT}/include/
```

Comment:
Arch linux adds a env-variable which can be parsed:

```
cat /etc/profile.d/petsc.sh 
export PETSC_DIR=/opt/petsc/linux-c-opt
```

Reply:
Added `$ENV{PETSC_DIR}/include/`. Please, check that `$ENV{PETSC_DIR}/lib/` points to the .so file.